File: farm-air-program-articles/src/main/java/com/example/articles/entity/Articles.java
package com.example.articles.entity;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.annotations.Mapper;

/**
 * <p>
 * 文章信息表
 * </p>
 *
 * @author linyunhong
 * @since 2024-11-26
 */
@Data
@Mapper
public class Articles implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 文章ID
     */
    private String id;

    /**
     * 分类ID
     */
    private Integer categoryId;

    /**
     * 文章分类名
     */
    private String categoryName;

    /**
     * 用户ID
     */
    private String userId;

    /**
     * 用户名
     */
    private String username;

    /**
     * 头像
     */
    private String avatarUrl;

    /**
     * 标题
     */
    private String title;

    /**
     * 封面
     */
    private String cover;


    /**
     * 内容
     */
    private String content;

    /**
     * 摘要
     */
    private String summary;

    /**
     * 发布时间
     */
    private LocalDateTime postTime;

    /**
     * 最后更新时间
     */
    private LocalDateTime lastUpdateTime;

    /**
     * 阅读数量
     */
    private Integer readCount;

    /**
     * 点赞数
     */
    private Integer goodCount;
    /**
     * 收藏数
     */
    private Integer collectionCount;

    /**
     * 评论数
     */
    private Integer commentCount;

    /**
     * 0未置顶 1:已置顶
     */
    private Integer topType;

    /**
     * -1已删除 0:待审核 1:已审核
     */
    private Integer status;

    public Articles() {
    }

    public Articles(String id, Integer categoryId, String categoryName, String userId, String username, String avatarUrl, String title, String cover, String content, String summary, LocalDateTime postTime, LocalDateTime lastUpdateTime, Integer readCount, Integer goodCount, Integer collectionCount, Integer commentCount, Integer topType, Integer status) {
        this.id = id;
        this.categoryId = categoryId;
        this.categoryName = categoryName;
        this.userId = userId;
        this.username = username;
        this.avatarUrl = avatarUrl;
        this.title = title;
        this.cover = cover;
        this.content = content;
        this.summary = summary;
        this.postTime = postTime;
        this.lastUpdateTime = lastUpdateTime;
        this.readCount = readCount;
        this.goodCount = goodCount;
        this.collectionCount = collectionCount;
        this.commentCount = commentCount;
        this.topType = topType;
        this.status = status;
    }
}

File: farm-air-program-web-vue3/src/api/articles.ts
// 使用axios发送请求
import axios from "axios";
import request from "@/api/utils/request";



// articleReqVo实体类
export interface Article {
    id: string;
    //作者信息也得在里面
    user: {
        id: string;
        username: string;
        nickname: string;
        avatar: string;
    };
    title: string;
    categoryId: number | null;
    categoryName: string;
    tags: string[];
    userId: string;
    username: string;
    avatarUrl: string;
    cover: string | null;
    content: string;
    summary: string | null;
    postTime: Date | null;
    lastUpdateTime: Date | null;
    readCount: number;
    goodCount: number;
    collectionCount: number;
    commentCount: number;
    topType: number;
    status: number;
    good: boolean;
    collection: boolean;
    highlightTitle: string;
    highlightContent: string;
}


export interface ArticlesColumns{
    id: string;
    title: string;
    categoryId: number | null;
    categoryName: string;
    userId: string;
    username: string;
    avatarUrl: string;
    cover: string | null;
    summary: string | null;
    postTime: Date | null;
    lastUpdateTime: Date | null;
    readCount: number;
    collectionCount: number;
    status: number;
    articles: [];
}


// 定义一个函数来发送 POST 请求
// export async function getUser(id: number): Promise<User> {
//     try {
//       const response = await axios.get(`/api/users/${id}`);
//       console.log('User created successfully:', response.data);
//       return response.data;
//     } catch (error) {
//       console.error('Error creating user:', error);
//       throw error;
//     }
//   }



export async function getArticleById(id: String){
    try {
      const response = await request.get(`/api/articles/articles/${id}`, {

      });
      console.log('articles fetched successfully:', response.data);
      return response.data;
    } catch (error) {
      console.error('Error fetching articles:', error);
      throw error;
    }
}

// 所有文章
export async function getArticles() {
  try {
    const response = await request.get(`/api/articles/articles`, {
      
    });
    console.log('articles fetched successfully:', response.data.data);
    return response.data; 
  } catch (error) {
    console.error('Error fetching user:', error);
    throw error;
  }
}



export async function createArticle(article:Article){
    try {
        const response = await request.post(`/api/articles/articles`, article);
        console.log('articles created successfully:', response.data);
        return response.data;
    } catch (error) {
        console.error('Error creating articles:', error);
        throw error;
    }
}
export async function getAllCategory() {
    try {
      const response = await request.get(`/api/articles/articlesCategory/all`);
      console.log('articles fetched successfully:', response.data);
      return response.data;
    } catch (error) {
        console.error('Error fetching articles:', error)
    }
}

//like表
export async function likeArticle(article: Article) {
    try {
        console.log("article",article)
      const response = await request.post(`/api/articles/articlesLike`, article);
      console.log('articles fetched successfully:', response.data);
      return response.data;
    } catch (error) {
      console.error('Error fetching articles:', error);
      throw error;
    }
  }

//普通搜索
export async function getSearchArticle(articleName:string, articleType:number){
    try{
        const response = await request.get(`/api/articles/articles/search`, {
            params: {
                articleName: articleName,
                status: articleType
            }
        });
        console.log('articles fetched successfully:', response.data);
        return response.data;
    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}
//es搜索 关键词
export async function getSearchArticles(keyword:string, page:number = 1, size:number = 5) {
    try{
        const response = await request.get(`/api/articles/articles/search/es`, {
            params: {
                keyword: keyword,
                page: page,
                size: size
            }
        });
        console.log('articles fetched successfully:', response.data.data);
        return response.data.data;
    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}


export async function createArticleColumns(articlesColumns:ArticleColumns){
    try{
        const response = await request.post(`/api/articles/articlesColumns`, articlesColumns);
        console.log('articles fetched successfully:', response.data);
        return response.data;

    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}
//查询一个用户创建的专栏列表

export async function getSearchColumns(columnsName:string, columnsType:number){
    try{
        const response = await request.get(`/api/articles/articlesColumns/search`, {
            params: {
                columnsName: columnsName,
                status: columnsType
            }
        });
        console.log('articles fetched successfully:', response.data);
        return response.data;
    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}

//查询一个分类的专栏列表
export async function getSearchColumnsByCategory(categoryId:number, status:number){
    try{
        const response = await request.get(`/api/articles/articlesColumns/searchByCategory`, {
            params: {
                categoryId: categoryId,
                status: status
            }
        });
        console.log('articles fetched successfully:', response.data);
        return response.data;
    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}
//查询一个专栏下的文章列表
export async function getSearchArticlesByColumns(columnsId:number, status:number){
    try{
        const response = await request.get(`/api/articles/articlesColumns/searchByColumns`, {
            params: {
                columnsId: columnsId,
                status: status
            }
        });
        console.log('articles fetched');
        return response.data;
    }catch (error){
        console.error('Error fetching articles:', error);
        throw error;
    }
}